core/extension/gdextension_interface.h

```diff
@@ -304,6 +306,12 @@ typedef struct {
 	GDExtensionClassCreateInstance create_instance_func; // (Default) constructor; mandatory. If the class is not instantiable, consider making it virtual or abstract.
 	GDExtensionClassFreeInstance free_instance_func; // Destructor; mandatory.
 	GDExtensionClassGetVirtual get_virtual_func; // Queries a virtual function by name and returns a callback to invoke the requested virtual function.
+	// Can be used instead of `get_virtual_func`. Returns user data that will be passed to `call_virtual_func`.
+	// Returning `NULL` from this function signals to Godot that the virtual function is not overridden.
+	// Data returned from this function should be managed by the extension and must be valid until the extension is deinitialized.
+	GDExtensionClassGetVirtuaCallData get_virtual_call_data_func;
+	// Used to call virtual functions when `get_virtual_call_data_func` is not null.
+	GDExtensionClassCallVirtualWithData call_virtual_func;
```

Comment:
What is missing is a brief guide regarding when to use the new approach (`get_virtual_call_data_func` + `call_virtual_func`) vs. when to use the old one (`get_virtual_func`). In particular, it should be mentioned that they both coexist with separate use cases, one doesn't supersede the other.

Also, maybe mention that the data returned by this function is passed back to `call_virtual_function`.

Last, typo `GDExtensionClassGetVirtuaCallData` -> "Virtua"

Reply:
Updated docs, let me know if that's clearer.

Reply:
This text is still the same?
```cpp
	// Can be used instead of `get_virtual_func`. Returns user data that will be passed to `call_virtual_func`.
	// Returning `NULL` from this function signals to Godot that the virtual function is not overridden.
	// Data returned from this function should be managed by the extension and must be valid until the extension is deinitialized.
	GDExtensionClassGetVirtualCallData get_virtual_call_data_func;
	// Used to call virtual functions when `get_virtual_call_data_func` is not null.
	GDExtensionClassCallVirtualWithData call_virtual_func;
```

Reply:
🤔 I don't think so? I'm using force pushes.... I didn't think I originally had the "Returns user data that will be passed to `call_virtual_func`" in the original documentation. But it also doesn't look as complete as I remember writing.

I'll take another stab at it.